Initial draft of upstream docs (#18)
che-incubator/kubernetes-image-puller: README.md

[![Build Status](https://ci.centos.org/job/devtools-kubernetes-image-puller-build-master/badge/icon)](https://ci.centos.org/job/devtools-kubernetes-image-puller-build-master/)

## Requirements
This is an upstream version of the [kubernetes-image-puller](https://github.com/redhat-developer/kubernetes-image-puller).  Where the downstream puller requires integrations with [fabric-oso-proxy](https://github.com/fabric8-services/fabric8-oso-proxy) and [fabric8-auth](https://github.com/fabric8-services/fabric8-auth), and impersonates users in multiple clusters, this application is meant to run on a single cluster, and pre-pull Eclipse Che images.

> To cache images, this app creates a Daemonset on the desired cluster, which in turn creates a pod on each node in the cluster consisting of a list of containers with command `sleep infinity`. This ensures that all nodes in the cluster have those images cached. We also periodically check the health of the daemonset and re-create it if necessary.

Reviewer: Technically speaking, `sleep 30d` (which we do) is more compatible than `sleep infinity`. IIRC, `sleep` in busybox doesn't support "infinity".

The application can be deployed via Helm or by processing and applying OpenShift Templates.

## Configuration

> Configuration is done via env vars pulled from `./deploy/helm/templates/configmap.yaml`, or `./deploy/openshift/configmap.yaml`, depending on the deployment method.

Reviewer: ```suggestion
Configuration is done via environment variables pulled from `./deploy/helm/templates/configmap.yaml`, or `./deploy/openshift/configmap.yaml`, depending on the deployment method.
```

The config values to be set are:

| Env Var | Usage | Default |
| -- | -- | -- |
| `CACHING_INTERVAL_HOURS` | Interval, in hours, between checking health of daemonsets | `"1"` |
| `CACHING_MEMORY_REQUEST` | The container memory request | `10Mi` |

> | `CACHING_MEMORY_LIMIT` | The container memory limit | `20Mi` |

Reviewer: Maybe something like "memory request(limit) for each cached image"

| `DAEMONSET_NAME`         | Name of daemonset to be created | `kubernetes-image-puller` |

> | `NAMESPACE`              | Namespace where daemonset is to be created. Shared for all users | `k8s-image-puller` |

Reviewer: "Shared for all users" only really makes sense downstream (where users ~ clusters)

| `IMAGES`                 | List of images to be cached, in the format `<name>=<image>;...` | Contains a default list of images, but should be configured when deploying |

> | `NODE_SELECTOR` | The node selector to run the daemonsets on particular nodes. | `'{}'` |

Reviewer: ```suggestion
| `NODE_SELECTOR` | Node selector applied to pods created by the daemonset. | `'{}'` |
```

Author: Thank you, I wasn't 100% sure what this did :D
